src/migen_axi/interconnect/axi2csr.py

```diff
@@ -7,19 +7,18 @@
 
 
 class AXI2CSR(Module):
-    def __init__(self, bus_axi=None, bus_csr=None, addr_extension=4):
+    def __init__(self, bus_axi=None, bus_csr=None, addr_extension=None):
         self.bus = bus_axi or axi.Interface()
         self.csr = bus_csr or csr_bus.Interface()
 
-        # internal bus that will connect to address decoder
-        # needs to be wider than CSR bus to accommodate both bus and any SRAM
-        # data width also identical to AXI to accomodate SRAM
-        self._internal_csr = csr_bus.Interface(
-            data_width=len(self.bus.r.data),
-            address_width=len(self.csr.adr) + addr_extension)
+        # if None, will be calculated automatically
+        # if a number, it will be added to csr bus width
```

Comment:
I would remove support for putting a fixed number entirely.

Reply:
ok done, in this case add_slave was also made a protected method (as in, name changed to _add_slave), so the chance of meddling with address space decreases further